jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java

```diff
@@ -213,15 +213,18 @@ public void testFlushWhenQueueFull() {
   public void testFlushUpdatesQueueLength() throws Exception {
     // change sender to blocking mode
     sender.permitAppend(0);
-    RemoteReporter remoteReporter = (RemoteReporter) reporter;
-    remoteReporter.flush();
+
+    await("flush() execution via TimerTask").atMost(5, TimeUnit.SECONDS).until(
+        () -> metricsReporter.gauges.get("jaeger.reporter-queue") != null
+    );
```

Comment:
TimerTask fires with 0 delay and `await()` waits just for that (until metric != null). It waits ~100ms because that's the default poll interval, that could be set to 5ms.

I wait for TimerTask to avoid having it call flush at some random time. Ideally I'd want to write just these:
1. add 10 spans
2. check that reporter-queue is still 0
3. flush()
4. check that it was updated and is no longer 0

But TimerTask first run (after ~0 delay) might happen between 1. and 2. and check in 2. would fail - I think that's why it is flaky right now (I'm not 100% sure, it doesn't fail for me). The same can happen in the scenario you described.
`awaiting` the first TimerTask run solves this.

I'll change that poll interval to 5ms and add only 3 spans. Let me know if I should rework it more.

Now, there is another test `testCloseWhenQueueFull` which takes 1 second - because it tests the case where sender blocks when reporter is closing and there's that 1sec `CLOSE_ENQUEUE_TIMEOUT_MILLIS`. I don't see an easy way to make that test fast other than by making this timeout configurable. Which might be a good idea...